### ISSUE TITLE
iCubTelemVizServer crashes occasionally if we close the web browser client (visualization tab)

### ISSUE THREAD

**Issue by nunoguedelha** [ami-iit/yarp-openmct#48]
## Reproducing the issue
- Run the visualization application as usual.
- Open the link `localhot:8080` as usual, on the browser, in a new tab.
- Close the tab.

## Issue
As we close the tab, the server `iCubTelemVizServer` crashes with the error:
```
/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/node_modules/express-ws/node_modules/ws/lib/WebSocket.js:219
    else throw new Error('not opened');
         ^

Error: not opened
    at WebSocket.send (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/node_modules/express-ws/node_modules/ws/lib/WebSocket.js:219:16)
    at notifySubscribers (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/realtime-server.js:22:20)
    at /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:146:9
    at Array.forEach (native)
    at ICubTelemetry.notify (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:145:20)
    at ICubTelemetry.generateTelemetry (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:137:10)
    at ICubTelemetry.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:53:18)
    at Array.forEach (native)
    at ICubTelemetry.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:52:33)
    at wrapper [as _onTimeout] (timers.js:275:19)
```

### Note

The issue occurs whether or not we close the child process `node server.js` which was running the Open-MCT Static server (directly from the terminal or through the main process `node iCubTelemVizServer.js`).

**Comment by nunoguedelha** [comment]
## The Fix

#### websocket
https://github.com/websockets/ws/blob/master/doc/ws.md
https://nicedoc.io/websockets/ws/blob/master/doc/ws.md

websocket.readyState : {Number} The current state of the connection. This is one of the ready state constants.
websocket.terminate(): Forcibly close the connection. Internally this calls socket.destroy().
websocket.onclose    : {Function} An event listener to be called when connection is closed. The listener receives a CloseEvent named "close".

<h3>Ready state constants</h3>

Constant | Value | Description
-- | -- | --
CONNECTING | 0 | The connection is not yet open.
OPEN | 1 | The connection is open and ready to communicate.
CLOSING | 2 | The connection is in the process of closing.
CLOSED | 3 | The connection is closed.

We should check that websocket.readyState === websocket.OPEN (= 1) before sending any data, in:

https://github.com/ami-iit/yarp-openmct/blob/2bd6bf5749d1313064fc706ec29c6f0eb205184c/iCubTelemVizServer/realtime-server.js#L20-L24

**Comment by nunoguedelha** [comment]
Fixed in #50
